runtime/startup_aux.c

```diff
@@ -103,34 +88,39 @@ static void scanmult (char_os *opt, uintnat *var)
 void caml_parse_ocamlrunparam(void)
 {
   char_os *opt = caml_secure_getenv (T("OCAMLRUNPARAM"));
-  uintnat p;
+
+  init_startup_params();
 
   if (opt == NULL) opt = caml_secure_getenv (T("CAMLRUNPARAM"));
 
   if (opt != NULL){
     while (*opt != '\0'){
       switch (*opt++){
-      case 'a': scanmult (opt, &caml_init_policy); break;
-      case 'b': scanmult (opt, &p); caml_record_backtraces(p); break;
-      case 'c': scanmult (opt, &p); caml_cleanup_on_exit = (p != 0); break;
-      case 'h': scanmult (opt, &caml_init_heap_wsz); break;
-      case 'H': scanmult (opt, &caml_use_huge_pages); break;
-      case 'i': scanmult (opt, &caml_init_heap_chunk_sz); break;
-      case 'l': scanmult (opt, &caml_init_max_stack_wsz); break;
-      case 'M': scanmult (opt, &caml_init_custom_major_ratio); break;
-      case 'm': scanmult (opt, &caml_init_custom_minor_ratio); break;
-      case 'n': scanmult (opt, &caml_init_custom_minor_max_bsz); break;
-      case 'o': scanmult (opt, &caml_init_percent_free); break;
-      case 'O': scanmult (opt, &caml_init_max_percent_free); break;
-      case 'p': scanmult (opt, &p); caml_parser_trace = (p != 0); break;
+      //case 'a': scanmult (opt, &p); caml_set_allocation_policy (p); break;
+      case 'b': scanmult (opt, &params.backtrace_enabled); break;
+      case 'c': scanmult (opt, &params.cleanup_on_exit); break;
+      case 'e': scanmult (opt, &params.eventlog_enabled); break;
+      case 'f': scanmult (opt, &params.init_fiber_wsz); break;
+      case 'h': scanmult (opt, &params.init_heap_wsz); break;
+      //case 'H': scanmult (opt, &caml_use_huge_pages); break;
+      case 'i': scanmult (opt, &params.init_heap_chunk_sz); break;
+      case 'l': scanmult (opt, &params.init_max_stack_wsz); break;
+      case 'M': scanmult (opt, &params.init_custom_major_ratio); break;
+      case 'm': scanmult (opt, &params.init_custom_minor_ratio); break;
+      case 'n': scanmult (opt, &params.init_custom_minor_max_bsz); break;
+      case 'o': scanmult (opt, &params.init_percent_free); break;
+      case 'O': scanmult (opt, &params.init_max_percent_free); break;
+      case 'p': scanmult (opt, &params.parser_trace); break;
       case 'R': break; /*  see stdlib/hashtbl.mli */
-      case 's': scanmult (opt, &caml_init_minor_heap_wsz); break;
-      case 't': scanmult (opt, &caml_trace_level); break;
-      case 'v': scanmult (opt, &caml_verb_gc); break;
-      case 'w': scanmult (opt, &caml_init_major_window); break;
+      case 's': scanmult (opt, &params.init_minor_heap_wsz); break;
+      case 't': scanmult (opt, &params.trace_level); break;
+      case 'v': scanmult (opt, &params.verb_gc); break;
+      case 'V': scanmult (opt, &params.verify_heap); break;
+      //case 'w': scanmult (opt, &caml_init_major_window); break;
       case 'W': scanmult (opt, &caml_runtime_warnings); break;
       case ',': continue;
       }
+      --opt; /* to handle patterns like ",b=1" */
```

Comment:
I don't understand this change. It seems to me that empty clauses are handled by line 121.

Reply:
git-blaming reveals that line 121 was added in in trunk in c0440d8161d, so what we have here is probably a merge of two solutions to the same problem.

Reply:
Agree there looks like two solutions merged together here. 

I've reverted the multicore introduced change in favour of the trunk solution in #10873